src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -286,7 +290,27 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
                 }
             }
 
-            return validTagName && validAttributeNames;
+            var validParentTagName = ValidateParentTagName(attribute.ParentTag, errorSink);
+
+            return validTagName && validAttributeNames && validParentTagName;
+        }
+
+        /// <summary>
+        /// Internal for unit testing.
```

Comment:
IMO these comments are kinda pointless. With the exception of shared code, we exclusively use `internal` method for InternalsVisibleTo unit testing. We should only be calling out the opposite case (when `internal` is used for non-test cases)